ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -81,25 +89,29 @@ public static SourceBuilder instance() {
     final boolean forceMaterialization = !planInfo.isRepartitionedInPlan(streamSource);
 
     final KTable<K, GenericRow> transformed = source.transformValues(
-        new AddKeyAndPseudoColumns<>(keyGenerator, streamSource.getPseudoColumnVersion()));
+        new AddPseudoColumnsToMaterialize<>(streamSource.getPseudoColumnVersion()));
+
+    final KTable<K, GenericRow> toPotentiallyMaterialize;
```

Comment:
```suggestion
    final KTable<K, GenericRow> maybeMaterialized;
```

Reply:
Committed, but can you explain why you think this is better?

Reply:
1. This variable is assigned after potential materialization, not before. `maybeMaterialized` conveys this, while `toPotentiallyMaterialize` conveys the opposite.
2. It's shorter :)

Reply:
Makes total sense in the context of your next comment. However, given how I changed the code to materialize on `transform`, I left it as `toPotentiallyMaterialize` (see next push)

Reply:
It looks like `toPotentiallyMaterialize` is still being set after the table might've already been materialized? I don't understand the change referred to in this comment:

> given how I changed the code to materialize on transform, I left it as toPotentiallyMaterialize

Reply:
I was reading it as "we potentially materialize this" when declaring the variable, not when assigning it. Thinking about it, I agree that naming based off of when it's assigned makes more sense.